### PR TITLE
Fix issues where we were comparing a struct to null.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph.cs [hunk 1]
```diff
@@ -249,7 +249,7 @@ private void ComputeReverseTransitiveReferences(ProjectId project, HashSet<Proje
         /// </summary>
         public IEnumerable<ProjectId> GetTopologicallySortedProjects(CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (_lazyTopologicallySortedProjects == null)
+            if (_lazyTopologicallySortedProjects.IsDefault)
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph.cs]: Oh goodness can you make a comment that we actually take advantage of this?

File: src/Workspaces/Core/Portable/Utilities/EventMap.cs [hunk 2]
```diff
@@ -160,10 +160,7 @@ internal EventHandlerSet(object registries)
                 _registries = (ImmutableArray<Registry<TEventHandler>>)registries;
             }
 
-            public bool HasHandlers
-            {
-                get { return _registries != null && _registries.Length > 0; }
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/Utilities/EventMap.cs]: Should we be checking for `IsDefaultOrEmpty` here?  See https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs#L176, but it looks like `Length` throws on default instances.

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs [hunk 3]
```diff
@@ -566,16 +566,13 @@ internal static class DependentTypeFinder
                 foreach (var task in tasks)
                 {
                     var result = await task.ConfigureAwait(false);
-                    if (result != null)
+                    foreach (var derivedType in result)
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs]: Check for default?

(See https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs#L1037)

**Pull request closed without merging**